dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -310,6 +310,8 @@ def _histogram_bin_error(self, input_array):
 
         inds = np.digitize(input_array, bin_edges)
 
+        inds = np.where(inds >= len(input_array), len(input_array) - 1, inds)
```

Comment:
Maybe, should the line `bin_edges[-1] += 1e-3` be relative to the scaling of the numbers? This line is supposed to catch said issue, but clearly it is not. 1e-3 maybe too small for the large numbers bc of overflow. Or is it an issue with merge and the min/max of the bins aren't properly being updated.

I do agree this would fix the issue, but I'm concerned that after merging, this histogram may be incorrectly binned.

Thoughts?

Reply:
The fix should work, I've double checked. @JGSweets : 1e-3 doesn't work for the large numbers as you said. And, this fix doesn't affect the merge/min/max.

Reply:
Correct, but is the code not working correctly because the bin edges are incorrect at min/max?

Reply:
The reason for using the +/- is because it is not expensive, but using the where operation would be on a large dataset, hence a fix not using where would be ideal. One that adjust the +/- relative to the bin.

But we first need to make sure the bins are correct after merge.

Reply:
Maybe do a swap, set temp value to current, set bin value to no.inf, do digitize, and finally swap back

Reply:
actually the check should be for length of `bin_edges` instead of `input_array`, i.e., should be
```
inds = np.where(inds >= len(bin_edges), len(bin_edges) - 1, inds)
```

Reply:
then, we can remove `bin_edges[-1] += 1e-3` before it and `bin_edges[-1] -= 1e-3` after it